handlers/whatsapp/whatsapp.go

```diff
@@ -611,12 +623,53 @@ func sendWhatsAppMsg(msg courier.Msg, sendPath *url.URL, token string, payload i
 		if rrCheck == nil {
 			elapsed := time.Now().Sub(start)
 			checkLog := courier.NewChannelLogFromError("unable to build contact check request", msg.Channel(), msg.ID(), elapsed, err)
-			return "", []*courier.ChannelLog{log, checkLog}, err
+			return "", "", []*courier.ChannelLog{log, checkLog}, err
 		}
 		checkLog := courier.NewChannelLogFromRR("Contact check", msg.Channel(), msg.ID(), rrCheck).WithError("Status Error", err)
 
 		if err != nil {
-			return "", []*courier.ChannelLog{log, checkLog}, err
+			return "", "", []*courier.ChannelLog{log, checkLog}, err
+		}
+		// update contact URN and msg destiny with returned wpp id
+		wppID, err := jsonparser.GetString(rrCheck.Body, "contacts", "[0]", "wa_id")
+
+		if err == nil {
+			var updatedPayload interface{}
+
+			// handle msg type casting
+			if text, ok := payload.(mtTextPayload); ok {
```

Comment:
I think this is probably best done with a type switch: 
  https://tour.golang.org/methods/16